---
title: "Sequence Analysis"
author: "Tung Trinh"
output: 
  word_document:
    toc: TRUE
---

```{r echo = T, warning=FALSE, message= FALSE}
# Clean up the R enviriment before conduct analysis
rm(list = ls())
# Load neccesary package 
library(reshape2) # reshape data structure
library(rentrez) # NCBI searching package
library(knitr) # to create nice table by using kable()
library(msa) # Write up fasta file
library(Biostrings) #
library(Peptides)
library(seqinr)
```

#**Sequence Analysis**
Progressive myoclonus epilepsies (PMEs) is a neural diseases that resulted from genetic disorders. Researching further about this disease is neccesary to develop gene edit therapies. Most of analysation below was run by R languega with some additional packages.

###**qRT-PCR analysis**
The qRT-PCR resuls consists of two groups healthy people (healthy) and PMEs (PME) patients with five different GoIs, **EPM2A**,**CSTB**,**SCN1A**,**NHLRC1** and **KCNC1**. Each of group also includes of housekeeper (HK) value that used for data normalisation. 
```{r echo = T, message=FALSE,warning=F}
# Data input with ordered values.
# These values is followed this order: Healthy, HK of Healthy, PME and HK of PME
EPM2A <- c(19.87, 10.63, 21.54, 11.27)
CSTB <- c(17.36, 10.21,21.92, 10.36)
SCN1A <- c(20.68, 10.21, 20.47, 10.01)
NHLRC1 <- c(18.79,10.08, 20.01, 10.58)
KCNC1 <- c(20.34, 10.58, 20.44, 11.1)
# Create data frame for more convinient calculation
qPCR<- data.frame(EPM2A, CSTB, SCN1A, NHLRC1, KCNC1)
print(qPCR)
# Calculate the Delta of healthy group
health_qPCR <- qPCR[1,] - qPCR[2,]
health_qPCR
# Calculate the Delta of PMEs group 
patient_pPCR <- qPCR[3,] - qPCR[4,]
patient_pPCR
# The difference of level of gene expression between two groups healthy and PMEs
2**-(patient_pPCR - health_qPCR)
```

##**Accessing NCBI database**
Due to large database of nucleotide data in NCBI bank, the combination of code-based search engine command and specific keywords help filtering, managing and process big data. The main purpose of this search is finding coding sequence of listed above gene which related to PMEs so we used **Gene Name** and **Organism** filters in nucleotide bank to archieve more specific result.

```{r, echo = T, warning= F, message = F}
# Search in nucleotide data bank of NCBI
EPM2A_search <- entrez_search(db = "nucleotide", term = "EPM2A[Gene Name] AND homo sapiens[Organism]")
# Extract data summary from NCBI by searched ID
EPM2A_sum <- entrez_summary(db ="nucleotide", id=EPM2A_search$ids)
```

There are **`r length(EPM2A_search$ids)`** found results for **EPM2A** gene in human. We should look at the summary to choose apporiaete sequence.

```{r, echo = T , warning=F, message= F}
# Using head() to show the first five results only
# Title
head(extract_from_esummary(EPM2A_sum, "title"))
# Sequence length
head(extract_from_esummary(EPM2A_sum,"slen"))
```

By looking at the title and the sequence length, `r extract_from_esummary(EPM2A_sum, "title")[3]` and `r extract_from_esummary(EPM2A_sum, "title")[4]` are the most apporiate data for **EPM2A** gene. Their accession numbers are shown below.
```{r, echo = T ,warning= F , message=F}
# Accession number
extract_from_esummary(EPM2A_sum,"caption")[c(3,4)]
# Chormosomal Location
extract_from_esummary(EPM2A_sum, c("subtype","subname"))[c(3,4)]
```
Both genes are on human chromosome 6 with sequence length of `r extract_from_esummary(EPM2A_sum,"slen")[1]` and `r extract_from_esummary(EPM2A_sum,"slen")[2]` bp, respectively. We repeat the same proceduce with other genes. Below is the summary table.

```{r, echo = F, warning=F, message=F}
#CSTB
CSTB_search <- entrez_search(db = "nucleotide", term = "CSTB[Gene Name] AND homo sapiens[Organism]")
CSTB_sum <- entrez_summary(db ="nucleotide", id=CSTB_search$ids)
# Accession number
extract_from_esummary(CSTB_sum,"caption")[c(1,2)]
# Sequence length
extract_from_esummary(CSTB_sum, "slen")[c(1,2)]
# Chormosomal Location
extract_from_esummary(CSTB_sum, c("subtype","subname"))[c(1,2)]

#SCN1A
SCN1A_search <- entrez_search(db = "nucleotide", term = "SCN1A[Gene Name] AND homo sapiens[Organism]")
SCN1A_sum <- entrez_summary(db ="nucleotide", id=SCN1A_search$ids)
# Accession number
extract_from_esummary(SCN1A_sum,"caption")[1]
# Sequence length
extract_from_esummary(SCN1A_sum, "slen")[1]
extract_from_esummary(SCN1A_sum, c("subtype", "subname"))[1]

# NHLRC1
NHLRC1_search <- entrez_search(db = "nucleotide", term = "NHLRC1[Gene Name] AND homo sapiens[Organism]")
NHLRC1_sum <- entrez_summary(db ="nucleotide", id=NHLRC1_search$ids)
#accession number
extract_from_esummary(NHLRC1_sum,"caption")[c(1,2)]
#sequence length
extract_from_esummary(NHLRC1_sum, "slen")[c(1,2)]

# KCNC1
KCNC1_search <- entrez_search(db = "nucleotide", term = "KCNC1[Gene Name] AND homo sapiens[Organism]")
KCNC1_sum <- entrez_summary(db ="nucleotide", id=KCNC1_search$ids)
#accession number
extract_from_esummary(KCNC1_sum,"caption")
#sequence length
extract_from_esummary(KCNC1_sum, "slen")

# Create data for summary
# Total results found
hit_found <- c(length(EPM2A_search$ids), length(CSTB_search$ids),length(SCN1A_search$ids),length(NHLRC1_search$ids),length(KCNC1_search))
# Full gene name
f_gene_name <- c(extract_from_esummary(EPM2A_sum,"title")[3],extract_from_esummary(CSTB_sum,"title")[1],extract_from_esummary(SCN1A_sum,"title")[3], extract_from_esummary(NHLRC1_sum,"title")[1], extract_from_esummary(KCNC1_sum,"title")[2])
# Accession number
accession_number <- c(extract_from_esummary(EPM2A_sum,"caption")[3],extract_from_esummary(CSTB_sum,"caption")[1],extract_from_esummary(SCN1A_sum,"caption")[3], extract_from_esummary(NHLRC1_sum,"caption")[1], extract_from_esummary(KCNC1_sum,"caption")[2])
# Sequence length
seq_length <- c(extract_from_esummary(EPM2A_sum,"slen")[3],extract_from_esummary(CSTB_sum,"slen")[1],extract_from_esummary(SCN1A_sum,"slen")[3], extract_from_esummary(NHLRC1_sum,"slen")[1], extract_from_esummary(KCNC1_sum,"slen")[2])
# Chromosomal Location
chromo_loc <- c(6,21,2,6,11)
dat <- data.frame(hit_found,f_gene_name,accession_number,seq_length,chromo_loc)
rownames(dat) <- c("EPM2A","CSTB","SCN1A","NHLRC1","KCNC1")
colnames(dat) <- c("Total results found","Full Gene Name","Accession Number","Sequence Length","Chromosomal Location")
kable(dat)
```

###**Statistical Test**
CSTB gene was used for further analysing. This section shows how significant level of PME patient with CSTB gene is, comparing with healthy group. Logarithm of 10 was applied for better visualization.  

```{r, echo = T , warning= F , message=FALSE}
# Create data frame
healthy <- c(33508,33508,26489,42152,25442,24899)
PME <- c(1152,3344,2811,3501,3653,3989)
gene_expression <- data.frame(healthy, PME)
gene_expression <- melt(gene_expression)
# Boxplot to see the distribution
boxplot(log10(value) ~ variable, data = gene_expression, xlab =  "Group", ylab = "Log10(value)")
# Exclude the outlier out of the data
PME <- PME[-1]
# Calculate standard deviation
sd_PME <- sd(PME, na.rm = T)
sd_healthy <- sd(healthy)
# Plot the result
data <- c(log10(mean(PME, na.rm = T)),log10(mean(healthy)))
barplot(height = data, ylim = c(0, 5), beside =  T, names.arg = c("PMEs patient","Healthy Group"), ylab = "log10 of value", col = c("lightblue","lightpink"))
#Draw standard error bar
arrows(0.7,log10(mean(PME, na.rm = T)+sd_PME),0.7,log10(mean(PME, na.rm = T)),1,90,3 )
arrows(1.9,log10(mean(healthy)+sd_healthy),1.9,log10(mean(healthy)),1,90,3 )
```

Student's t-test is the most simplest stastistical test which we can determine the significant level of difference between two groups of data. 
```{r, echo = T, warning= F, message= F}
# Two indepenent samples T-test
# Compare the variance and mean of 2 sample populations are equal or not
t.test(healthy,PME, alternative = "two.sided")
```

To check the result with other statistical test, we can perform linear model with analysis of variance (ANOVA). The result also show the significant difference two groups.
```{r,echo = T, warning = F , message= F}
anova(lm(value ~ variable, data = gene_expression))
```


##**DNA Alignment**
```{r, echo = F,eval = F, message=F, warning= F}
# The gene sequence can be downloaded by using rentrez package
# Create sequence fasta file 
# Write fasta file
#Healthy person with allele 1 and 2
health_a1 <- "CTGCAGCGCTGCGCCCCGCCCTCCCTCGTGCGCGGCCCGGACCCCGCCACCCTGCAGGATTGCCCCTACTCCGACTGCCCCTTCCCTATCGTCCCACCCTGCGCGCCCAACCCACCGGCGACACCCGGCCGCGCCCCCGCCCCGGTCCGTGTGACTCGGCGCCCGGAAAGACGATACCAGCCCCGGGAGGGGGGCGCTCCCTCCCGACACCAGCGCTGGGCGCGGAGACCCAGCCTGCGGCGAGTGGTGGCCAGGCTCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCGTCCCTTCTTGCGGGGCCACCGCGACCCCGCAGGGGACTCCGAAGCCAAAGTGCCTCCTCCCCGCCCCTTGGTTCCGCCCGCGCGTCACGTGACCCCAGCGCCTACTTGGGCTGAGGAGCCGCCGCGTCCCCTCGCCGAGTCCCCTCGCCAGATTCCCTCCGTCGCCGCCAAGATGATGTGCGGGGCGCCCTCCGCCACGCAGCCGGCCACCGCCGAGACCCAGCACATCGCCGACCAG"
health_a2 <- "CTGCAGCGCTGCGCCCCGCCCTCCCTCGTGCGCGGCCCGGACCCCGCCACCCTGCAGGATTGCCCCTACTCCGACTGCCCCTTCCCTATCGTCCCACCCTGCGCGCCCAACCCACCGGCGACACCCGGCCGCGCCCCCGCCCCGGTCCGTGTGACTCGGCGCCCGGAAAGACGATACCAGCCCCGGGAGGGGGGCGCTCCCTCCCGACACCAGCGCTGGGCGCGGAGACCCAGCCTGCGGCGAGTGGTGGCCAGGCTCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCGTCCCTTCTTGCGGGGCCACCGCGACCCCGCAGGGGACTCCGAAGCCAAAGTGCCTCCTCCCCGCCCCTTGGTTCCGCCCGCGCGTCACGTGACCCCAGCGCCTACTTGGGCTGAGGAGCCGCCGCGTCCCCTCGCCGAGTCCCCTCGCCAGATTCCCTCCGTCGCCGCCAAGATGATGTGCGGGGCGCCCTCCGCCACGCAGCCGGCCACCGCCGAGACCCAGCACATCGCCGACCAG"
# Patient A with allele 1 and 2
pat_A1 <- "CTGCAGCGCTGCGCCCCGCCCTCCCTCGTGCGCGGCCCGGACCCCGCCACCCTGCAGGATTGCCCCTACTCCGACTGCCCCTTCCCTATCGTCCCACCCTGCGCGCCCAACCCACCGGCGACACCCGGCCGCGCCCCCGCCCCGGTCCGTGTGACTCGGCGCCCGGAAAGACGATACCAGCCCCGGGAGGGGGGCGCTCCCTCCCGACACCAGCGCTGGGCGCGGAGACCCAGCCTGCGGCGAGTGGTGGCCAGGCTCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCGTCCCTTCTTGCGGGGCCACCGCGACCCCGCAGGGGACTCCGAAGCCAAAGTGCCTCCTCCCCGCCCCTTGGTTCCGCCCGCGCGTCACGTGACCCCAGCGCCTACTTGGGCTGAGGAGCCGCCGCGTCCCCTCGCCGAGTCCCCTCGCCAGATTCCCTCCGTCGCCGCCAAGATGATGTGCGGGGCGCCCTCCGCCACGCAGCCGGCCACCGCCGAGACCCAGCACATCGCCGACCAG"
pat_A2 <- "CTGCAGCGCTGCGCCCCGCCCTCCCTCGTGCGCGGCCCGGACCCCGCCACCCTGCAGGATTGCCCCTACTCCGACTGCCCCTTCCCTATCGTCCCACCCTGCGCGCCCAACCCACCGGCGACACCCGGCCGCGCCCCCGCCCCGGTCCGTGTGACTCGGCGCCCGGAAAGACGATACCAGCCCCGGGAGGGGGGCGCTCCCTCCCGACACCAGCGCTGGGCGCGGAGACCCAGCCTGCGGCGAGTGGTGGCCAGGCTCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCGTCCCTTCTTGCGGGGCCACCGCGACCCCGCAGGGGACTCCGAAGCCAAAGTGCCTCCTCCCCGCCCCTTGGTTCCGCCCGCGCGTCACGTGACCCCAGCGCCTACTTGGGCTGAGGAGCCGCCGCGTCCCCTCGCCGAGTCCCCTCGCCAGATTCCCTCCGTCGCCGCCAAGATGATGTGCGGGGCGCCCTCCGCCACGCAGCCGGCCACCGCCGAGACCCAGCACATCGCCGACCAG"
#Patient B with allele 1 and 2
pat_B1 <- "CTGCAGCGCTGCGCCCCGCCCTCCCTCGTGCGCGGCCCGGACCCCGCCACCCTGCAGGATTGCCCCTACTCCGACTGCCCCTTCCCTATCGTCCCACCCTGCGCGCCCAACCCACCGGCGACACCCGGCCGCGCCCCCGCCCCGGTCCGTGTGACTCGGCGCCCGGAAAGACGATACCAGCCCCGGGAGGGGGGCGCTCCCTCCCGACACCAGCGCTGGGCGCGGAGACCCAGCCTGCGGCGAGTGGTGGCCAGGCTCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCGTCCCTTCTTGCGGGGCCACCGCGACCCCGCAGGGGACTCCGAAGCCAAAGTGCCTCCTCCCCGCCCCTTGGTTCCGCCCGCGCGTCACGTGACCCCAGCGCCTACTTGGGCTGAGGAGCCGCCGCGTCCCCTCGCCGAGTCCCCTCGCCAGATTCCCTCCGTCGCCGCCAAGATGATGTGCGGGGCGCCCTCCGCCACGCAGCCGGCCACCGCCGAGACCCAGCACATCGCCGACCAG"
pat_B2 <- "CTGCAGCGCTGCGCCCCGCCCTCCCTCGTGCGCGGCCCGGACCCCGCCACCCTGCAGGATTGCCCCTACTCCGACTGCCCCTTCCCTATCGTCCCACCCTGCGCGCCCAACCCACCGGCGACACCCGGCCGCGCCCCCGCCCCGGTCCGTGTGACTCGGCGCCCGGAAAGACGATACCAGCCCCGGGAGGGGGGCGCTCCCTCCCGACACCAGCGCTGGGCGCGGAGACCCAGCCTGCGGCGAGTGGTGGCCAGGCTCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCGTCCCTTCTTGCGGGGCCACCGCGACCCCGCAGGGGACTCCGAAGCCAAAGTGCCTCCTCCCCGCCCCTTGGTTCCGCCCGCGCGTCACGTGACCCCAGCGCCTACTTGGGCTGAGGAGCCGCCGCGTCCCCTCGCCGAGTCCCCTCGCCAGATTCCCTCCGTCGCCGCCAAGATGATGTGCGGGGCGCCCTCCGCCACGCAGCCGGCCACCGCCGAGACCCAGCACATCGCCGACCAG"
#Patient C with allele 1 and 2
pat_C1 <- "CTGCAGCGCTGCGCCCCGCCCTCCCTCGTGCGCGGCCCGGACCCCGCCACCCTGCAGGATTGCCCCTACTCCGACTGCCCCTTCCCTATCGTCCCACCCTGCGCGCCCAACCCACCGGCGACACCCGGCCGCGCCCCCGCCCCGGTCCGTGTGACTCGGCGCCCGGAAAGACGATACCAGCCCCGGGAGGGGGGCGCTCCCTCCCGACACCAGCGCTGGGCGCGGAGACCCAGCCTGCGGCGAGTGGTGGCCAGGCTCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCGTCCCTTCTTGCGGGGCCACCGCGACCCCGCAGGGGACTCCGAAGCCAAAGTGCCTCCTCCCCGCCCCTTGGTTCCGCCCGCGCGTCACGTGACCCCAGCGCCTACTTGGGCTGAGGAGCCGCCGCGTCCCCTCGCCGAGTCCCCTCGCCAGATTCCCTCCGTCGCCGCCAAGATGATGTGCGGGGCGCCCTCCGCCACGTAGCCGGCCACCGCCGAGACCCAGCACATCGCCGACCAG"
pat_C2 <- "CTGCAGCGCTGCGCCCCGCCCTCCCTCGTGCGCGGCCCGGACCCCGCCACCCTGCAGGATTGCCCCTACTCCGACTGCCCCTTCCCTATCGTCCCACCCTGCGCGCCCAACCCACCGGCGACACCCGGCCGCGCCCCCGCCCCGGTCCGTGTGACTCGGCGCCCGGAAAGACGATACCAGCCCCGGGAGGGGGGCGCTCCCTCCCGACACCAGCGCTGGGCGCGGAGACCCAGCCTGCGGCGAGTGGTGGCCAGGCTCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCGTCCCTTCTTGCGGGGCCACCGCGACCCCGCAGGGGACTCCGAAGCCAAAGTGCCTCCTCCCCGCCCCTTGGTTCCGCCCGCGCGTCACGTGACCCCAGCGCCTACTTGGGCTGAGGAGCCGCCGCGTCCCCTCGCCGAGTCCCCTCGCCAGATTCCCTCCGTCGCCGCCAAGATGATGTGCGGGGCGCCCTCCGCCGCAGCCGGCCACCGCCGAGACCCAGCACATCGCCGACCAG"
#Create fasta file
write.fasta(as.list(c(health_a1,health_a2,pat_A1,pat_A2,pat_B1,pat_B2,pat_C1,pat_C2)),as.list(c("Health_1","Health_2","Patient_A_1","Patient_A_2","Patient_B_1","Patient_B_2","Patient_C_1","Patient_C_2")), "data/sequence.fa", open = "w+")
```
DNA alignment analysis was conducted by using ClustalW method in R program language. The implement code and result are shown below.
```{r,echo = T, warning= F , message=F}
# Read data sequence
# The data sequence file was created by R, the code can be found in supplement script
sequence <- readDNAStringSet("data/sequence.fa")
# DNA alignment with ClustalW method
DNA_alignment <- msa(sequence, method = "ClustalW", type = "dna", order ="input")
# Visualize the result
# Please note that order to produce pdf type, you have to install Latex beforehand
msaPrettyPrint(DNA_alignment, output = "pdf", showNames = "left", askForOverwrite = F)
# The pdf file is then converted into image file and inserted into document
```

###**Protein Translation**
Open Reading Frame (ORF) should be taken into consideration. Three ORFs,starts from position 0, +1 and +2, are translated into amino acid. The translated amino acid sequences are then used for protein alignment.
```{r ,echo = T, warning=FALSE, message=FALSE}
#Prepare open reading frame from position 0 for protein translation
sequence_ORF0 <- sequence
#Prepare open reading frame from position +1 for protein translation
sequence_ORF1 <- DNAStringSet(sequence, start = 2)
#Prepare open reading frame +from position 2 for protein translation
sequence_ORF2 <- DNAStringSet(sequence, start = 3)
#Translation ORF0
ORF0 <- Biostrings::translate(sequence_ORF0)
#Translation ORF1
ORF1 <- Biostrings::translate(sequence_ORF1)
#Translation ORF2 
ORF2 <- Biostrings::translate(sequence_ORF2)
```

###**Protein Alignment**
ClustalW method is also applied for protein alignment in this section.
```{r, echo = T, warning= F , message=F}
# Protein alignment
# Note: to be able to use msaPrettyPrint, you need to install laxtex platform in your computer
# Protein alignment for Open reading frame 0
aligned_ORF0 <- msa(ORF0, method = "ClustalW", type ="protein", order = "input")
msaPrettyPrint(aligned_ORF0, output = "pdf", showNames = "left", askForOverwrite = F)
# Protein alignment for Open reading frame 1
aligned_ORF1 <- msa(ORF1, method = "ClustalW", order ="input", type = "protein")
msaPrettyPrint(aligned_ORF1,output = "pdf", showNames = "left", showLogo = "none", askForOverwrite = F)
# Protein alignment for Open reading frame 2
aligned_ORF2 <- msa(ORF2, method = "ClustalW", type = "protein", order = "input")
msaPrettyPrint(aligned_ORF2, output = "pdf", showNames = "left", askForOverwrite = F)
# The pdf file is  then converted into image file and inserted into document
```

##**Protein properties prediction** 
```{r, echo = T , warning= F , message=F}
# Amino acid composition for healthy person, patienct A and B
aaComp(ORF2[1:6])
# Molecular weight for healthy person, patienct A and B
mw(ORF2[1:6])
# Aliphatic index of protein for healthy person, patienct A and B
aIndex(ORF2[1:6])
# Boman index (potential protein interation) for health person, patienct A and B
boman(ORF2[1:6])
# Amino acid composition for healthy person and patient C
aaComp(ORF2[c(1,2,7,8)])
# Molecular weight for healthy person and patient C
mw(ORF2[c(1,2,7,8)])
# Aliphatic index of protein for healthy person and pateint C
aIndex(ORF2[c(1,2,7,8)])
# Boman index (potential protein interation) for health person and patient C
boman(ORF2[c(1,2,7,8)])
```
