rm(list = ls())

#Loading necessary package

#source("http://www.bioconductor.org/biocLite.R")
#biocLite("msa")
# biocLite("DECIPHER")
#install.packages("Biostrings")
#install.packages("seqinr")
#install.packages("bio3d")
# install.packages("rentrez")
library(rentrez)
library(msa)
# library(bio3d)
library(Biostrings)
# library(seqinr)
library(Peptides)

#Write fasta file
#Healthy person with allele 1 and 2
# health_a1 <- "CTGCAGCGCTGCGCCCCGCCCTCCCTCGTGCGCGGCCCGGACCCCGCCACCCTGCAGGATTGCCCCTACTCCGACTGCCCCTTCCCTATCGTCCCACCCTGCGCGCCCAACCCACCGGCGACACCCGGCCGCGCCCCCGCCCCGGTCCGTGTGACTCGGCGCCCGGAAAGACGATACCAGCCCCGGGAGGGGGGCGCTCCCTCCCGACACCAGCGCTGGGCGCGGAGACCCAGCCTGCGGCGAGTGGTGGCCAGGCTCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCGTCCCTTCTTGCGGGGCCACCGCGACCCCGCAGGGGACTCCGAAGCCAAAGTGCCTCCTCCCCGCCCCTTGGTTCCGCCCGCGCGTCACGTGACCCCAGCGCCTACTTGGGCTGAGGAGCCGCCGCGTCCCCTCGCCGAGTCCCCTCGCCAGATTCCCTCCGTCGCCGCCAAGATGATGTGCGGGGCGCCCTCCGCCACGCAGCCGGCCACCGCCGAGACCCAGCACATCGCCGACCAG"
# health_a2 <- "CTGCAGCGCTGCGCCCCGCCCTCCCTCGTGCGCGGCCCGGACCCCGCCACCCTGCAGGATTGCCCCTACTCCGACTGCCCCTTCCCTATCGTCCCACCCTGCGCGCCCAACCCACCGGCGACACCCGGCCGCGCCCCCGCCCCGGTCCGTGTGACTCGGCGCCCGGAAAGACGATACCAGCCCCGGGAGGGGGGCGCTCCCTCCCGACACCAGCGCTGGGCGCGGAGACCCAGCCTGCGGCGAGTGGTGGCCAGGCTCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCGTCCCTTCTTGCGGGGCCACCGCGACCCCGCAGGGGACTCCGAAGCCAAAGTGCCTCCTCCCCGCCCCTTGGTTCCGCCCGCGCGTCACGTGACCCCAGCGCCTACTTGGGCTGAGGAGCCGCCGCGTCCCCTCGCCGAGTCCCCTCGCCAGATTCCCTCCGTCGCCGCCAAGATGATGTGCGGGGCGCCCTCCGCCACGCAGCCGGCCACCGCCGAGACCCAGCACATCGCCGACCAG"
# Patient A with allele 1 and 2
# pat_A1 <- "CTGCAGCGCTGCGCCCCGCCCTCCCTCGTGCGCGGCCCGGACCCCGCCACCCTGCAGGATTGCCCCTACTCCGACTGCCCCTTCCCTATCGTCCCACCCTGCGCGCCCAACCCACCGGCGACACCCGGCCGCGCCCCCGCCCCGGTCCGTGTGACTCGGCGCCCGGAAAGACGATACCAGCCCCGGGAGGGGGGCGCTCCCTCCCGACACCAGCGCTGGGCGCGGAGACCCAGCCTGCGGCGAGTGGTGGCCAGGCTCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCGTCCCTTCTTGCGGGGCCACCGCGACCCCGCAGGGGACTCCGAAGCCAAAGTGCCTCCTCCCCGCCCCTTGGTTCCGCCCGCGCGTCACGTGACCCCAGCGCCTACTTGGGCTGAGGAGCCGCCGCGTCCCCTCGCCGAGTCCCCTCGCCAGATTCCCTCCGTCGCCGCCAAGATGATGTGCGGGGCGCCCTCCGCCACGCAGCCGGCCACCGCCGAGACCCAGCACATCGCCGACCAG"
# pat_A2 <- "CTGCAGCGCTGCGCCCCGCCCTCCCTCGTGCGCGGCCCGGACCCCGCCACCCTGCAGGATTGCCCCTACTCCGACTGCCCCTTCCCTATCGTCCCACCCTGCGCGCCCAACCCACCGGCGACACCCGGCCGCGCCCCCGCCCCGGTCCGTGTGACTCGGCGCCCGGAAAGACGATACCAGCCCCGGGAGGGGGGCGCTCCCTCCCGACACCAGCGCTGGGCGCGGAGACCCAGCCTGCGGCGAGTGGTGGCCAGGCTCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCGTCCCTTCTTGCGGGGCCACCGCGACCCCGCAGGGGACTCCGAAGCCAAAGTGCCTCCTCCCCGCCCCTTGGTTCCGCCCGCGCGTCACGTGACCCCAGCGCCTACTTGGGCTGAGGAGCCGCCGCGTCCCCTCGCCGAGTCCCCTCGCCAGATTCCCTCCGTCGCCGCCAAGATGATGTGCGGGGCGCCCTCCGCCACGCAGCCGGCCACCGCCGAGACCCAGCACATCGCCGACCAG"
#Patient B with allele 1 and 2
# pat_B1 <- "CTGCAGCGCTGCGCCCCGCCCTCCCTCGTGCGCGGCCCGGACCCCGCCACCCTGCAGGATTGCCCCTACTCCGACTGCCCCTTCCCTATCGTCCCACCCTGCGCGCCCAACCCACCGGCGACACCCGGCCGCGCCCCCGCCCCGGTCCGTGTGACTCGGCGCCCGGAAAGACGATACCAGCCCCGGGAGGGGGGCGCTCCCTCCCGACACCAGCGCTGGGCGCGGAGACCCAGCCTGCGGCGAGTGGTGGCCAGGCTCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCGTCCCTTCTTGCGGGGCCACCGCGACCCCGCAGGGGACTCCGAAGCCAAAGTGCCTCCTCCCCGCCCCTTGGTTCCGCCCGCGCGTCACGTGACCCCAGCGCCTACTTGGGCTGAGGAGCCGCCGCGTCCCCTCGCCGAGTCCCCTCGCCAGATTCCCTCCGTCGCCGCCAAGATGATGTGCGGGGCGCCCTCCGCCACGCAGCCGGCCACCGCCGAGACCCAGCACATCGCCGACCAG"
# pat_B2 <- "CTGCAGCGCTGCGCCCCGCCCTCCCTCGTGCGCGGCCCGGACCCCGCCACCCTGCAGGATTGCCCCTACTCCGACTGCCCCTTCCCTATCGTCCCACCCTGCGCGCCCAACCCACCGGCGACACCCGGCCGCGCCCCCGCCCCGGTCCGTGTGACTCGGCGCCCGGAAAGACGATACCAGCCCCGGGAGGGGGGCGCTCCCTCCCGACACCAGCGCTGGGCGCGGAGACCCAGCCTGCGGCGAGTGGTGGCCAGGCTCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCGTCCCTTCTTGCGGGGCCACCGCGACCCCGCAGGGGACTCCGAAGCCAAAGTGCCTCCTCCCCGCCCCTTGGTTCCGCCCGCGCGTCACGTGACCCCAGCGCCTACTTGGGCTGAGGAGCCGCCGCGTCCCCTCGCCGAGTCCCCTCGCCAGATTCCCTCCGTCGCCGCCAAGATGATGTGCGGGGCGCCCTCCGCCACGCAGCCGGCCACCGCCGAGACCCAGCACATCGCCGACCAG"
#Patient C with allele 1 and 2
# pat_C1 <- "CTGCAGCGCTGCGCCCCGCCCTCCCTCGTGCGCGGCCCGGACCCCGCCACCCTGCAGGATTGCCCCTACTCCGACTGCCCCTTCCCTATCGTCCCACCCTGCGCGCCCAACCCACCGGCGACACCCGGCCGCGCCCCCGCCCCGGTCCGTGTGACTCGGCGCCCGGAAAGACGATACCAGCCCCGGGAGGGGGGCGCTCCCTCCCGACACCAGCGCTGGGCGCGGAGACCCAGCCTGCGGCGAGTGGTGGCCAGGCTCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCGTCCCTTCTTGCGGGGCCACCGCGACCCCGCAGGGGACTCCGAAGCCAAAGTGCCTCCTCCCCGCCCCTTGGTTCCGCCCGCGCGTCACGTGACCCCAGCGCCTACTTGGGCTGAGGAGCCGCCGCGTCCCCTCGCCGAGTCCCCTCGCCAGATTCCCTCCGTCGCCGCCAAGATGATGTGCGGGGCGCCCTCCGCCACGTAGCCGGCCACCGCCGAGACCCAGCACATCGCCGACCAG"
# pat_C2 <- "CTGCAGCGCTGCGCCCCGCCCTCCCTCGTGCGCGGCCCGGACCCCGCCACCCTGCAGGATTGCCCCTACTCCGACTGCCCCTTCCCTATCGTCCCACCCTGCGCGCCCAACCCACCGGCGACACCCGGCCGCGCCCCCGCCCCGGTCCGTGTGACTCGGCGCCCGGAAAGACGATACCAGCCCCGGGAGGGGGGCGCTCCCTCCCGACACCAGCGCTGGGCGCGGAGACCCAGCCTGCGGCGAGTGGTGGCCAGGCTCCCCGCCCCGCGCCCCGCCCCGCGCCCCGCCCCGCGCGTCCCTTCTTGCGGGGCCACCGCGACCCCGCAGGGGACTCCGAAGCCAAAGTGCCTCCTCCCCGCCCCTTGGTTCCGCCCGCGCGTCACGTGACCCCAGCGCCTACTTGGGCTGAGGAGCCGCCGCGTCCCCTCGCCGAGTCCCCTCGCCAGATTCCCTCCGTCGCCGCCAAGATGATGTGCGGGGCGCCCTCCGCCGCAGCCGGCCACCGCCGAGACCCAGCACATCGCCGACCAG"
# write.fasta(as.list(c(health_a1,health_a2,pat_A1,pat_A2,pat_B1,pat_B2,pat_C1,pat_C2)),as.list(c("Health_1","Health_2","Patient_A_1","Patient_A_2","Patient_B_1","Patient_B_2","Patient_C_1","Patient_C_2")), "data/sequence.fa", open = "w+")

#Load fasta file 
sequence <- readDNAStringSet("data/sequence.fa")
sequence_ORF0 <- sequence
sequence_ORF1 <- DNAStringSet(sequence, start = 2)
sequence_ORF2 <- DNAStringSet(sequence, start = 3)
#Translate DNA seq to protein seq with 3 possible Open Reading Frame
ORF0 <- Biostrings::translate(sequence_ORF0)
ORF1 <- Biostrings::translate(sequence_ORF1)
ORF2 <- Biostrings::translate(sequence_ORF2)
# seqinr::write.fasta(as.list(ORF0),as.list(names(ORF0)), "data/ORF0.fa", open = "w+b")
#msa 
aalignment<- sequence[c(1,2,7,8)]
aDNA_alignment <- msa(aalignment, method = "ClustalW", type = "dna", order ="input")
msaPrettyPrint(aDNA_alignment, output = "pdf", showNames = "left", askForOverwrite = F)

DNA_alignment<- msa(sequence, method ="ClustalW", type ="dna", order = "input")
msaPrettyPrint(DNA_alignment , output = "pdf", askForOverwrite = F)
#Alignment
aligned_ORF0 <- msa(ORF0, method = "ClustalW", type ="protein", order = "input")
msaPrettyPrint(aligned_ORF0, output = "pdf", showNames = "left", askForOverwrite = F)

aligned_ORF1 <- msa(ORF1, method = "ClustalW", order ="input", type = "protein")
msaPrettyPrint(aligned_ORF1,output = "pdf", showNames = "left", showLogo = "none", askForOverwrite = F)

aligned_ORF2 <- msa(ORF2, method = "ClustalW", type = "protein", order = "input")
msaPrettyPrint(aligned_ORF2, output = "pdf", showNames = "left", askForOverwrite = F)
#
aacomp(ORF2)
mw(ORF2)
aindex(ORF2)
boman(ORF2)

#Part 1 
healthy <- c(33508,33508,26489,42152,25442,24899)
PME <- c(1152,3344,2811,3501,3653,3989)

gene_expression <- data.frame(healthy, PME)
library(reshape2)
gene_expression <- melt(gene_expression)
graphics::boxplot(log10(value) ~ variable, data = gene_expression, xlab= "Group", ylab = "log10 of value")
# boxplot(value ~ variable, data = gene_expression)
anova(lm(value ~ variable, data = gene_expression, test = "Chisq"))

# one outier in PME 
mean(PME[-1])
mean(healthy)
round(mean(healthy)/mean(PME[-1]), digits = 2)
